polar-core/src/parser.rs

```diff
@@ -32,37 +32,31 @@ pub enum Line {
 }
 
 fn to_parse_error(e: ParseError<usize, lexer::Token, error::ParseError>) -> error::ParseError {
+    use error::ParseError::*;
+
     match e {
-        ParseError::InvalidToken { location: loc } => error::ParseError::InvalidToken { loc },
-        ParseError::UnrecognizedEOF { location: loc, .. } => {
-            error::ParseError::UnrecognizedEOF { loc }
-        }
+        ParseError::InvalidToken { location: loc } => InvalidToken { loc },
+        ParseError::UnrecognizedEOF { location: loc, .. } => UnrecognizedEOF { loc },
         ParseError::UnrecognizedToken {
             token: (loc, t, _), ..
         } => match t {
-            Token::Debug | Token::Cut | Token::In | Token::New => error::ParseError::ReservedWord {
+            Token::Debug | Token::Cut | Token::In | Token::New => ReservedWord {
                 token: t.to_string(),
                 loc,
             },
-            _ => error::ParseError::UnrecognizedToken {
+            _ => UnrecognizedToken {
                 token: t.to_string(),
                 loc,
             },
         },
-        ParseError::ExtraToken { token: (loc, t, _) } => error::ParseError::ExtraToken {
+        ParseError::ExtraToken { token: (loc, t, _) } => ExtraToken {
             token: t.to_string(),
             loc,
         },
         ParseError::User { error } => error,
     }
 }
 
-pub fn parse_term(src: &str) -> PolarResult<Term> {
-    polar::TermParser::new()
-        .parse(0, Lexer::new(src))
-        .map_err(|e| to_parse_error(e).into())
-}
```

Comment:
This is the only actual change in this file. We were only using this function in the `tests` mod down below, so I merged it with `tests::parse_term`, which was previously calling this function.